addons/website_version/controllers/main.py

```diff
@@ -0,0 +1,155 @@
+        mod_version = request.env['website_version.version']
+        version = mod_version.browse(version_id)
+        name_list = []
+        for view in version.view_ids:
+            name_list.append(view.name)
+        return name_list
+
+    @http.route(['/website_version/google_access'], type='json', auth="user")
+    def google_authorize(self, **kw):
+        #Check if client_id and client_secret are set to get the authorization from Google
+        gs_obj = request.env['google.service']
+        gm_obj = request.env['google.management']
+
+        client_id = gs_obj.get_client_id('management', context=kw.get('local_context'))
+        client_secret = gs_obj.get_client_secret('management', context=kw.get('local_context'))
+        if not client_id or not client_secret:
+            dummy, action = request.registry.get('ir.model.data').get_object_reference(request.cr, request.uid, 'website_version', 'action_config_settings_google_management')
+            return {
+                "status": "need_config_from_admin",
+                "url": '',
+                "action": action
+            }
+        url = gm_obj.authorize_google_uri(from_url=kw.get('fromurl'), context=kw.get('local_context'))
+        return {
+            "status": "need_auth",
+            "url": url
+        }
+
+    @http.route(['/website_version/set_google_access'], type='json', auth="user", website=True)
+    def set_google_access(self, ga_key, view_id, client_id, client_secret):
+        #To set ga_key, view_id, client_id, client_secret
+        website_id = request.context.get('website_id')
+        web = request.env['website'].browse(website_id)
+        web.write({'google_analytics_key': ga_key.strip(), 'google_analytics_view_id': view_id.strip()})
+        if client_id and client_secret:
+            icp = request.env['ir.config_parameter']
+            icp.set_param('google_management_client_id', client_id.strip() or '', groups=['base.group_system'])
+            icp.set_param('google_management_client_secret', client_secret.strip() or '', groups=['base.group_system'])
+
+    @http.route(['/website_version/all_versions_all_goals'], type='json', auth="user", website=True)
+    def all_versions_all_goals(self, view_id):
+        #To get all versions and all goals to create an experiment
+        view = request.env['ir.ui.view']
+        version = request.env['website_version.version']
+        goal = request.env['website_version.goals']
+        icp = request.env['ir.config_parameter']
+        v = view.browse(view_id)
+        website_id = request.website.id
+        tab_version = version.search_read([('website_id', '=', website_id), '|', ('view_ids.key', '=', v.key), ('view_ids.key', '=', 'website.footer_default')], ['id', 'name'])
+        tab_goal = goal.search_read([], ['id', 'name'])
+        #Check if all the parameters are set to communicate with Google analytics
+        if icp.get_param('google_management_token'):
+            check_conf = 2
```

Comment:
Another example of naming. You assign 0, 1 or 2 to check_conf, that is then used to set config of create_experiment. In the template, you do if config == 1, ... . Unless I reverse engineer the whole code, I can't understand the various configurations. Maybe use constants, strings, and/or put a docstring somewhere, here and in create_experiment for example.